src/context/providers/PointsGridFormDataProvider.js

```diff
@@ -129,59 +221,57 @@ export class PointsGridFormDataProvider extends mix(JSONSchemaFormDataProvider).
         };
     }
 
-    get _defaultData() {
-        return {
-            dimensions: this._defaultDimensions,
-            shifts: this._defaultShifts,
-            KPPRA: this._defaultKPPRA,
-            preferKPPRA: false,
-            reciprocalVectorRatios: this.reciprocalVectorRatios,
-        };
-    }
-
-    get _defaultDataWithMaterial() {
-        // if `data` is present and material is updated, prioritize `data` when `preferKPPRA` is not set
-        return this.preferKPPRA
-            ? this._getGridFromKPPRA(this.KPPRA)
-            : this.data || this._defaultData;
-    }
-
-    get defaultData() {
-        return this.material ? this._defaultDataWithMaterial : this._defaultData;
-    }
-
-    _getGridFromKPPRA(KPPRA) {
-        const reciprocalLattice = new Made.ReciprocalLattice(this.material.lattice);
+    _getDimensionsFromKPPRA(KPPRA) {
         const nAtoms = this.material ? this.material.Basis.nAtoms : 1;
-        return {
-            dimensions: reciprocalLattice.getDimensionsFromPoints(KPPRA / nAtoms),
-            shifts: this._defaultShifts,
-        };
+        return this.reciprocalLattice.getDimensionsFromPoints(KPPRA / nAtoms);
     }
 
-    _getKPPRAFromGrid(grid = this.defaultData) {
+    _getKPPRAFromDimensions(dimensions) {
         const nAtoms = this.material ? this.material.Basis.nAtoms : 1;
-        return grid.dimensions.reduce((a, b) => a * b) * nAtoms;
+        return dimensions.reduce((a, b) => a * b) * nAtoms;
     }
 
     static _canTransform(data) {
         return (
-            (data.preferKPPRA && data.KPPRA) ||
-            (!data.preferKPPRA && data.dimensions.every((d) => typeof d === "number"))
+            (data.preferGridMetric && data.gridMetricType && data.gridMetricValue) ||
+            (!data.preferGridMetric && data.dimensions.every((d) => typeof d === "number"))
         );
     }
 
+    calculateDimensions({ gridMetricType, gridMetricValue }) {
+        switch (gridMetricType) {
+            case "KPPRA":
+                return this._getDimensionsFromKPPRA(gridMetricValue);
+            case "spacing":
+                return this.reciprocalLattice.getDimensionsFromSpacing(gridMetricValue, "angstrom");
```

Comment:
Can we avoid passing the unit name explicitly and include as default in the function definition, maybe? Here nad line 258?

Reply:
`getDimensionsFromSpacing` already uses `ATOMIC_COORD_UNITS.cartesian` as default which also seems to be the default unit for other functions of `ReciprocalLattice`. So for consistency, I would leave the default units as is. In order to avoid mistyped units, however, I will replace the string with the corresponding enum.